source/common/event/dispatcher_impl.cc

```diff
@@ -37,6 +37,7 @@ DispatcherImpl::DispatcherImpl(Buffer::FactoryPtr&& factory)
 DispatcherImpl::~DispatcherImpl() {}
 
 void DispatcherImpl::clearDeferredDeleteList() {
+  ASSERT(run_tid_ == 0 || run_tid_ == Thread::Thread::currentThreadId());
```

Comment:
Added comment to explain (we sometimes are doing these ops from tests which don't `run()`). I've added the ASSERT to most other places modulo https://github.com/lyft/envoy/issues/1294.